src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,578 @@
+    ITradeModule tradeModule;
+    IRfqModule rfqModule;
+    IOptionAsset optionAsset;
+    ISpotFeed baseFeed;
+    PPTSAParams ppParams;
+    CollateralManagementParams collateralManagementParams;
+    bool isCallSpread;
+    bool isLongSpread;
+    /// @dev Only one hash is considered valid at a time, and it is revoked when a new one comes in.
+    bytes32 lastSeenHash;
+  }
+
+  struct StrikeData {
+    uint strike;
+    uint expiry;
+    int markPrice;
+    uint tradePrice;
+    int tradeAmount;
+  }
+
+  // keccak256(abi.encode(uint256(keccak256("lyra.storage.PrincipalProtectedTSA")) - 1)) & ~bytes32(uint256(0xff))
+  bytes32 private constant PPTSAStorageLocation = 0x273fadd8b9727c49bd8eabddca41871e1e7fc468f40ece57b00a1bd25002e500;
+
+  function _getPPTSAStorage() private pure returns (PPTSAStorage storage $) {
+    assembly {
+      $.slot := PPTSAStorageLocation
+    }
+  }
+
+  constructor() {
+    _disableInitializers();
+  }
+
+  function initialize(
+    address initialOwner,
+    BaseTSA.BaseTSAInitParams memory initParams,
+    PPTSAInitParams memory ppInitParams
+  ) external reinitializer(2) {
+    __BaseTSA_init(initialOwner, initParams);
+
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    $.depositModule = ppInitParams.depositModule;
+    $.withdrawalModule = ppInitParams.withdrawalModule;
+    $.rfqModule = ppInitParams.rfqModule;
+    $.tradeModule = ppInitParams.tradeModule;
+    $.optionAsset = ppInitParams.optionAsset;
+    $.baseFeed = ppInitParams.baseFeed;
+    $.isCallSpread = ppInitParams.isCallSpread;
+    $.isLongSpread = ppInitParams.isLongSpread;
+    BaseTSAAddresses memory tsaAddresses = getBaseTSAAddresses();
+
+    tsaAddresses.depositAsset.approve(address($.depositModule), type(uint).max);
+  }
+
+  ///////////
+  // Admin //
+  ///////////
+  function setPPTSAParams(CollateralManagementParams memory newCollateralMgmtParams, PPTSAParams memory pptsaParams)
+    external
+    onlyOwner
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    if (
+      pptsaParams.minSignatureExpiry < 1 minutes || pptsaParams.minSignatureExpiry > pptsaParams.maxSignatureExpiry
+        || newCollateralMgmtParams.worstSpotBuyPrice < 1e18 || newCollateralMgmtParams.worstSpotBuyPrice > 1.2e18
+        || newCollateralMgmtParams.worstSpotSellPrice > 1e18 || newCollateralMgmtParams.worstSpotSellPrice < 0.8e18
+        || newCollateralMgmtParams.spotTransactionLeniency < 1e18
+        || newCollateralMgmtParams.spotTransactionLeniency > 1.2e18
+        || pptsaParams.minMarkValueToStrikeDiffRatio > pptsaParams.maxMarkValueToStrikeDiffRatio
+        || pptsaParams.maxMarkValueToStrikeDiffRatio > 1e18 || pptsaParams.maxMarkValueToStrikeDiffRatio < 1e16
+        || pptsaParams.strikeDiff > 2e21 || pptsaParams.strikeDiff < 1e20
+        || pptsaParams.optionMaxTimeToExpiry <= pptsaParams.optionMinTimeToExpiry
+        || newCollateralMgmtParams.feeFactor > 0.05e18 || pptsaParams.maxTotalCostTolerance < 2e17
```

Comment:
like mentioned before, we need another param `pptsaParams.rfqFeeFactor`. Would double check the bounds with Vlad